benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/JumpTableMultipleEntryBenchmark.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    public class JumpTableMultipleEntryBenchmark
+    {
+        private string[] _strings;
+        private PathSegment[] _segments;
+        private int[] _samples;
+
+        private JumpTable _linearSearch;
+        private JumpTable _dictionary;
+
+        // All factors of 100 to support sampling
+        [Params(2, 5, 10, 25, 50, 100)]
+        public int Count;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _strings = GetStrings(100);
+            _segments = new PathSegment[100];
+
+            for (var i = 0; i < _strings.Length; i++)
+            {
+                _segments[i] = new PathSegment(0, _strings[i].Length);
+            }
+
+            _samples = new int[Count];
+            for (var i = 0; i < _samples.Length; i++)
+            {
+                _samples[i] = i * (_strings.Length / Count);
+            }
+
+            var entries = new List<(string text, int _)>();
+            for (var i = 0; i < _samples.Length; i++)
+            {
+                entries.Add((_strings[_samples[i]], i));
+            }
+
+            _linearSearch = new LinearSearchJumpTable(0, -1, entries.ToArray());
+            _dictionary = new DictionaryJumpTable(0, -1, entries.ToArray());
+        }
+
+        // This baseline is similar to SingleEntryJumpTable. We just want
+        // something stable to compare against.
+        [Benchmark(Baseline = true, OperationsPerInvoke = 100)]
+        public int Baseline()
+        {
+            var strings = _strings;
+            var segments = _segments;
+
+            var index = 0;
+            for (var i = 0; i < strings.Length; i++)
+            {
+                index = segments[i].Length == 0 ? -1 :
+                    segments[i].Length != strings[i].Length ? 1 :
+                    string.Compare(
+                        strings[i],
+                        segments[i].Start,
+                        strings[i],
+                        0,
+                        strings[i].Length,
+                        StringComparison.OrdinalIgnoreCase);
```

Comment:
This is a little obtuse but it basically simulates the minimum amount of work that an implementation will do.